### PR TITLE
Fixes for MemTree namespace nodes

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/1620

I also checked that this PR keeps the same DOM Test Suite score that we have for eXist 3.6.0 - https://github.com/exist-db/exist-domts#exist-db-360

### CODE REVIEW

**Comment by dizzzz** [comment]
@adamretter great work, super fast! Question: does this address the double-dash issue as well?

`xml

`

**Comment by adamretter** [comment]
@dizzzz Ah no. I missed that second issue in amongst the talk of the first one. I just added a follow-up in the issue for you for some more info ;-)

File: src/org/exist/dom/memtree/MemTreeBuilder.java [hunk 1]
```diff
@@ -147,15 +147,15 @@ public int startElement(final QName qname, final Attributes attributes) {
 
             // parse attributes
             for(int i = 0; i < attributes.getLength(); i++) {
```
Comment (dizzzz) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: probably we could use another loop, without the "i" index. A simple for loop over the attributes?

File: src/org/exist/dom/memtree/SAXAdapter.java [hunk 2]
```diff
@@ -125,7 +125,7 @@ public void startElement(final String namespaceURI, final String localName, fina
 
         for(int i = 0; i < atts.getLength(); i++) {
             if(atts.getQName(i).startsWith(XMLConstants.XMLNS_ATTRIBUTE)) {
-                final String prefix = atts.getLocalName(i);
+                final String prefix = null;
                 final String uri = atts.getValue(i);
                 if(namespaces == null || !namespaces.containsKey(prefix)) {
```
Comment (dizzzz) [inline src/org/exist/dom/memtree/SAXAdapter.java]: doesn't this yield into a NPE in line  130?

File: src/org/exist/dom/memtree/SAXAdapter.java [hunk 3]
```diff
@@ -125,7 +125,7 @@ public void startElement(final String namespaceURI, final String localName, fina
 
         for(int i = 0; i < atts.getLength(); i++) {
```
Comment (dizzzz) [inline src/org/exist/dom/memtree/SAXAdapter.java]: what about a Java5 loop on Attributes without 'i' variable?

**Pull request merged**